geps/gep-1619.md

```diff
@@ -163,43 +163,258 @@ Java application servers such as Tomcat and Jetty, were the first to standardize
-- SameSite=_[Strict|Lax|None]_ ([Draft RFC Update](https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-cookie-same-site-00))
-- Partitioned ([Draft RFC Update](https://www.ietf.org/archive/id/draft-cutler-httpbis-partitioned-cookies-01.html))
-
-**Session Cookie Strategy**
-
-Session cookie strategy is configuration specific to the creation of the cookie by the dataplane. These are not specific configuration parameters of the cookie itself, but instead the operation to take when inserting the cookie such as:
-- Rewriting an existing cookie value
-- Inserting a new cookie
-- Prefixing an existing cookie
-- Preserving an existing cookie value
-- Only adding cookies on specific HTTP methods (POST/GET)
+Other cookie attributes not defined by RFC6265, but are captured in draft RFCs and could be considered de facto
+standards due to wide acceptance are:
+- SameSite=[Strict|Lax|None]
+- Partitioned
+
+We implement core support for the `Name`, `Path`, and `TTL` attributes as the metrics in [implementations](#Implementations)
+for session persistence shows these being the most widely adopted and supported. Subsequently, we proceed with `Domain`,
+`SameSite`, `HttpOnly`, and `Secure` as extended support, considering their partial acceptance among APIs. The
+`Partitioned` attribute lacks support in any implementation API and will be omitted. Similarly, since the `Expires`
+attribute is redundant compared to the preferred `Max-Age` attribute, it will also be skipped.
+
+```go
+// SessionPersistencePolicy provides a way to define session persistence rules
+// for a service.
+//
+// Support: Core
+type SessionPersistencePolicy struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+    
+    // Spec defines the desired state of SessionPersistencePolicy.
+    Spec SessionPersistencePolicySpec `json:"spec"`
+    
+    // Status defines the current state of SessionPersistencePolicy.
+    Status SessionPersistencePolicyStatus `json:"status,omitempty"`
+}
+
+// SessionPersistencePolicySpec defines the desired state of
+// SessionPersistencePolicy.
+// Note: there is no Override or Default policy configuration.
+type SessionPersistencePolicySpec struct {
+    // TargetRef identifies an API object to apply policy to.
+    // Services are the only valid API target references.
+    TargetRef gatewayv1a2.PolicyTargetReference `json:"targetRef"`
+    
+    // HttpCookie defines and configures a cookie to achieve
+    // session persistence. 
+    HttpCookie *HttpCookie `json:"httpCookie"`
+}
+
+// HttpCookie defines a cookie to achieve session persistence.
+//
+// Support: Core
+type HttpCookie struct {
+    // Name defines the cookie's name.
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MaxLength=4096
+    Name String `json:"name,omitempty"`
+    
+    // Path defines the cookie's Path attribute.
+    //
+    // Support: Core
+    //
+    // +optional
```

Comment:
If this is optional what's the default value? Is it `/` ?

Reply:
I was thinking whether the Path should be part of the cookie `HttpCookie` configuration or should it be derived from the xRoute to which the config is attached (when it is attached to a xRoute). For example, if the request is to the path `/part1/part2/part3` and because the config in the HttpRoute resource specified cookie based session persistence to the path `/part1/part2` then the Path in the Cookie will automatically be `/part1/part2` . Is it not? Of course for config applied to Services the answer is not obvious (may be it is `/` or the default as you mentioned)

Reply:
Yea it's a interesting thought; especially if HTTPCookie attributes, i.e. path, are not configurable like Costin suggesting. Then it seems like it would be helpful to derive it.

However, thinking through the edge cases: 
1. What if you have an xRoute that matches all paths? Would that just be `/`?
2. What if an xRoute that matches multiple paths? A different `set-cookie` parameter would be derived for each path in the `xRoute`?
3. What if the path matching is using a [regex](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.PathMatchType)? How would you derive a path for the cookie with a regex?

Are you thinking this would be suggestion to implementors or a mandate?

Reply:
> Yea it's a interesting thought; especially if HTTPCookie attributes, i.e. path, are not configurable like Costin suggesting. Then it seems like it would be helpful to derive it.

Yes, Costin and I had an offline discussion on this which is where I got the idea.

> 
> However, thinking through the edge cases:
> 
> 1. What if you have an xRoute that matches all paths? Would that just be `/`?

Yes.

> 2. What if an xRoute that matches multiple paths? A different `set-cookie` parameter would be derived for each path in the `xRoute`?

Correct. The path that matched (in the xRoute) would be the value of `Path` in the cookie

> 3. What if the path matching is using a [regex](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.PathMatchType)? How would you derive a path for the cookie with a regex?

That's interesting. We can then use the longest prefix of the path that is not a regex. e.g. if the path is `/p1/p2/*/p3` and the request path was `/p1/p2/foo/p3` then the cookie path would be `/p1/p2` 

> 
> Are you thinking this would be suggestion to implementors or a mandate?

I am thinking a mandate rather than a suggestion

Reply:
Ack. Added updates to reflect `Path` auto-configuration. Please take a look.

Reply:
Resolving, I think I addressed this, but I still welcome more discussion my recent changes.